paparazzi/paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -114,7 +108,7 @@ class PaparazziPlugin : Plugin<Project> {
         task.compileSdkVersion.set(android.compileSdkVersion())
         task.mergeAssetsOutput.set(mergeAssetsOutputDir)
         task.paparazziResources.set(project.layout.buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.txt"))
-        task.snapshotDirectory.set(snapshotOutputDir)
+        task.snapshotDirectory.set(extension.snapshotDirectory.map { it.asFile.path })
```

Comment:
Does this need a relativize?, think about cross-machine cacheability.
Let's say we have
```
paparazzi {
    snapshotDirectory.set(file("foo/bar"))
}
```
someone (CI) runs the tasks on one machine and populates the remote cache.

someone else (dev) runs the task on their machine and pulls from remote cache. Given an exact same state the task should be up to date, but this only happens if the path is relative to the project.
```

Reply:
yeah, something like that. Can you please check what the other files in the same prepare task output file are? Because if they're absolute (hope not), then this is moot.

Reply:
If the snapshots should be inside the project or at least the rootProject directory, this should be  documented on the extension property (user contract).

Reply:
> Looks like with some of the changes we are creating a circular dependency

I'm a bit confused about how this is set up. The snapshots are written by record, and read by verify, but both of those are running INSIDE `testDebugUnitTest`.

So the snapshot dir is both input and output?

Reply:
But the weird part is we haven't changed it, and it was working, why it is failing now?

Reply:
We changed it:
```
    * What went wrong:
    Circular dependency between the following tasks:
    :compileDebugUnitTestJavaWithJavac
    +--- :compileDebugUnitTestKotlin
    |    \--- :preparePaparazziDebugResources <--------------- added a new input
    |         \--- :testDebugUnitTest <----------------------- which is registered as an output here?
    |              +--- :compileDebugUnitTestJavaWithJavac (*)
    |              \--- :compileDebugUnitTestKotlin (*)
    \--- :preparePaparazziDebugResources (*)
```
but the input is a string, so Gradle can't know, can it?

Reply:
I mean, this circular dependency is not happening on the first commit I did on this PR which already has those inputs, right?

Reply:
Might be because of the lazyness, the `get().asFile` in the original state broke the dependency. Silly thing, but could you try to swap the providers?

```
project.layout.projectDirectory.map { extension.snapshotDirectory.asFile.relativeTo(it.asFile).path }
```

Reply:
`projectDirectory` is not mappable. I will try to create a provider directly

Reply:
also give it a go without relativize, maybe it works that way 🤔